### PR TITLE
responsive views

### PR DESCRIPTION
**Pull request by SektorDV**

### CODE REVIEW

**Comment by stared** [comment]
I get a TypeScript error:

```
ERROR in /Users/pmigdal/my_repos/QG/quantum-game-2/src/components/Board/index.vue
137:37 Property 'boardScaler' does not exist on type '{ 'gridWrapper': HTMLElement; }'.
    135 |
    136 |   assessTileSize(): void {
  > 137 |     const currentWidth = this.$refs.boardScaler.getBoundingClientRect().width;
        |                                     ^
    138 |     const currentHeight = this.$refs.gridWrapper.getBoundingClientRect().height;
    139 |     this.$data.scalerStyle = {
    140 |       transform: `scale(${currentWidth / 870})`
ERROR in /Users/pmigdal/my_repos/QG/quantum-game-2/src/store/userStore.ts
```

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,4 @@ yarn-error.log*
 *.njsproj
 *.sln
 *.sw?
+src/config/firebase.ts
```
Comment (stared) [inline .gitignore]: Ignoring a code file creates non-obvious errors. If there are any credentials, then it should be some `secret.json` or `.config` or some other file having ONLY the core sensitive data.

File: src/components/Board/index.vue [hunk 2]
```diff
@@ -84,6 +87,15 @@ import SpeechBubble from '@/components/SpeechBubble.vue';
...
 export default class Board extends Vue {
+  data() {
+    return {
+      scalerStyle: {
+        transform: `scale(1)`,
+      },
+      boardHeight: 640
+      
+    }
+  }
```
Comment (stared) [inline src/components/Board/index.vue]: When using Vue TypeScript syntax, wouldn't:

```
scalerStyle = {
  transform: 'scale(1)',
}
boardHeight = 640
```
be more idiomatic?

File: src/components/Board/index.vue [hunk 3]
```diff
@@ -122,8 +134,13 @@ export default class Board extends Vue {
...
   assessTileSize(): void {
-    // const currentWidth = this.$refs.grid.getBoundingClientRect().width;
-    // this.tileSize = currentWidth / this.grid.cols;
+    const currentWidth = this.$refs.boardScaler.getBoundingClientRect().width;
+    const currentHeight = this.$refs.gridWrapper.getBoundingClientRect().height;
+    this.$data.scalerStyle = {
+      transform: `scale(${currentWidth / 870})`
+    }
+    this.$data.boardHeight = currentHeight;
+    //this.tileSize = currentWidth / this.grid.cols;
```
Comment (stared) [inline src/components/Board/index.vue]: Comment code should be avoided unless strictly needed (and with a comment WHY it was commented out).

File: src/components/Board/index.vue [hunk 4]
```diff
@@ -1,62 +1,65 @@
...
-        :particle="particle"
-        :animate="2"
-        :margin="2"
-        :display-magnetic="false"
-        :display-electric="true"
-        :display-gaussian="true"
-        :sigma="0.25"
+<div ref="boardScaler" class="board_scaler" :style="{height: boardHeight + 'px'}">
+  
+    <svg ref="gridWrapper" :style="scalerStyle" class="grid" :width="totalWidth" :height="totalHeight">
```
Comment (stared) [inline src/components/Board/index.vue]: `yarn lint` to conform with the indention styles

File: src/components/GamePage/GameKet.vue [hunk 5]
```diff
@@ -54,11 +55,23 @@ import QuantumFrame from '@/engine/QuantumFrame';
...
 })
 export default class GameKet extends Vue {
+  data() {
+    return {
+      ketHidden: true
+    }
+  }
+
+
+
```
Comment (stared) [inline src/components/GamePage/GameKet.vue]: After props, `ketHidden = true`. (TS syntax)

File: src/config/firebase.ts [hunk 6]
```diff
@@ -15,15 +15,15 @@ const firebaseConfig = {
...
+// const db = firebase.firestore();
+// const auth = firebase.auth();
+// auth.onAuthStateChanged((user) => {
+//   if (user) {
+//     $userStore.dispatch('FETCH_USER', user);
+//   }
+// });
 
-export { db, auth };
+// export { db, auth };
```
Comment (stared) [inline src/config/firebase.ts]: This change (plus, adding it to `.gitignore`) creates a few errors. Are you sure you wanted to make it?

**Comment by stared** [comment]
Overall, it is very nice, thank you @SektorDV ! There are some problems that need to be addressed, though (look at my comments).

I created, and merged, a different PR #92 where I addressed the most important things (I `data()`).

File: src/components/Board/index.vue (hunk 2 above)
Comment (SektorDV) [reply src/components/Board/index.vue]: that's possible, I'm not used to working with TS in Vue, so most of the changes I implemented use regular Vue.js syntax

File: src/components/Board/index.vue (hunk 3 above)
Comment (SektorDV) [reply src/components/Board/index.vue]: This and the neighboring lines were commented out from the beginning. I actually *uncommented* one of the lines. I could remove any commented lines I encounter in the future, if you like.

File: src/config/firebase.ts (hunk 6 above)
Comment (SektorDV) [reply src/config/firebase.ts]: that's an overlook on my end. I added that to circumvent firebase errors I've been getting that resulted in the app not loading altogether. I didn't intend to actually commit this (or the gitignore) change.

**Pull request closed without merging**